mlflow/models/cli.py

```diff
@@ -155,6 +155,35 @@ def prepare_env(
     ).prepare_env(model_uri=model_uri)
 
 
+@commands.command("generate-dockerfile")
+@cli_args.MODEL_URI_BUILD_DOCKER
+@click.option("--output-directory",
+              "-d",
+              default="generate-dockerfile-output",
+              help="output directory for generating dockerfile")
+@cli_args.ENV_MANAGER
+@cli_args.MLFLOW_HOME
+@cli_args.INSTALL_MLFLOW
+@cli_args.ENABLE_MLSERVER
+def generate_dockerfile(model_uri, output_directory, env_manager, mlflow_home, install_mlflow, enable_mlserver):
+    """
+    Generates a dockerfile whose default entrypoint serves an MLflow model at port 8080, using the
+    python_function flavor.
+
+    Essentially, it generates a dockerfile, which would've been the base for the docker image in `build_docker`
+    function
+    """
+    env_manager = env_manager or _EnvManager.CONDA
+    backend = _get_flavor_backend(model_uri, docker_build=True, env_manager=env_manager)
+    backend.generate_dockerfile(
```

Comment:
Can we check `backend.can_build_image()` and throw if the backend cannot build an image, which indicates that we cannot build a dockerfile for it? Can we add a similar check to https://github.com/mlflow/mlflow/blob/9ea22cdf4d18204e6b92585a028d40341667ac6e/mlflow/models/cli.py#L209?

Reply:
Added in both cases :+1: